vertx-web-client/src/main/java/io/vertx/webclient/BodyCodec.java

```diff
@@ -0,0 +1,169 @@
+   *
+   * @param encoding the encoding
+   * @return the codec
+   */
+  static BodyCodec<String> string(String encoding) {
+    return BodyCodecImpl.string(encoding);
+  }
+
+  /**
+   * @return the {@link Buffer} codec
+   */
+  static BodyCodec<Buffer> buffer() {
+    return BodyCodecImpl.BUFFER;
+  }
+
+  /**
+   * @return the {@link JsonObject} codec
+   */
+  static BodyCodec<JsonObject> jsonObject() {
+    return BodyCodecImpl.JSON_OBJECT;
+  }
+
+  /**
+   * Create and return a codec for Java objects encoded using Jackson mapper.
+   *
+   * @return a codec for mapping POJO to Json
+   */
+  @GenIgnore
+  static <U> BodyCodec<U> json(Class<U> type) {
+    return BodyCodecImpl.json(type);
+  }
+
+  /**
+   * @return a codec that simply discards the response
+   */
+  static BodyCodec<Void> none() {
+    return BodyCodecImpl.NONE;
+  }
+
+  /**
+   * Create a codec that buffers the entire body and then apply the {@code decode} function and returns the result.
+   *
+   * @param decode the decode function
+   * @return the created codec
+   */
+  static <T> BodyCodec<T> create(Function<Buffer, T> decode) {
+    return new BodyCodecImpl<>(decode);
+  }
+
+  /**
+   * A body codec that writes the body to a write stream.
+   *
+   * @param stream the destination tream
+   * @return the body codec for a write stream
+   */
+  static BodyCodec<Void> writeStream(WriteStream<Buffer> stream) {
+    return new BodyCodec<Void>() {
+      @Override
+      public void writeStream(Handler<AsyncResult<BodyStream<Void>>> handler) {
+        handler.handle(Future.succeededFuture(new BodyStream<Void>() {
+
+          Future<Void> fut = Future.future();
+
+          @Override
+          public Future<Void> state() {
+            return fut;
+          }
+
+          @Override
+          public void handle(Throwable cause) {
+            if (!fut.isComplete()) {
+              fut.fail(cause);
+            }
+          }
+
+          @Override
+          public WriteStream<Buffer> exceptionHandler(Handler<Throwable> handler) {
+            stream.exceptionHandler(handler);
+            return this;
+          }
+
+          @Override
+          public WriteStream<Buffer> write(Buffer data) {
+            stream.write(data);
+            return this;
+          }
+
+          @Override
+          public void end() {
+            stream.end();
+            if (!fut.isComplete()) {
+              fut.complete();
+            }
+          }
+
+          @Override
+          public WriteStream<Buffer> setWriteQueueMaxSize(int maxSize) {
+            return this;
+          }
+
+          @Override
+          public boolean writeQueueFull() {
+            return stream.writeQueueFull();
+          }
+
+          @Override
+          public WriteStream<Buffer> drainHandler(Handler<Void> handler) {
+            stream.drainHandler(handler);
+            return this;
+          }
+        }));
+      }
+    };
+  }
+
+  /**
+   * Reserved for internal usage.
```

Comment:
Well it could be called by normal users but there is no point to. I renamed the method to `create` and changed the doc to: _This method is usually called for creating the pump for the HTTP response and should not be called directly._